dso-l2/src/main/java/com/tc/l2/logging/BufferingAppender.java

```diff
@@ -18,43 +18,32 @@
  */
 package com.tc.l2.logging;
 
-import org.apache.log4j.Appender;
-import org.apache.log4j.AppenderSkeleton;
-import org.apache.log4j.spi.LoggingEvent;
+import ch.qos.logback.core.Appender;
+import ch.qos.logback.core.ConsoleAppender;
 
-import java.util.concurrent.ArrayBlockingQueue;
-import java.util.concurrent.BlockingQueue;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
 
 /**
  * An {@link Appender} that simply buffers records (in a bounded queue) until they're needed. This is used for making
  * sure all logging information gets to the file; we buffer records created before logging gets sent to a file, then
  * send them there.
  */
-public class BufferingAppender extends AppenderSkeleton {
+public class BufferingAppender<E> extends ConsoleAppender<E> {
 
-  private final BlockingQueue<LoggingEvent> buffer;
-  private boolean             on;
+  private final Queue<E> buffer;
+  private boolean on;
 
-  public BufferingAppender(int maxCapacity) {
-    this.buffer = new ArrayBlockingQueue<LoggingEvent>(maxCapacity);
-    this.on = true;
-  }
-
-  @Override
-  protected synchronized void append(LoggingEvent event) {
-    if (on) {
-      this.buffer.offer(event);
-    }
-  }
 
-  @Override
-  public boolean requiresLayout() {
-    return false;
+  public BufferingAppender() {
+    this.buffer = new ConcurrentLinkedQueue<>();
+    this.on = true;
   }
 
   @Override
-  public void close() {
-    // nothing needs to be here.
+  public void doAppend(E eventObject) {
+    buffer.add(eventObject);
+    super.doAppend(eventObject);
```

Comment:
This appender is an extension of ConsoleAppender. So while the buffering is happening on bootstrap we send those to the console as well.